src/lib.rs

```diff
@@ -226,6 +237,26 @@ impl CommitDomain {
             .map(|p| p + Wnaf::new().scalar(r).base(self.R))
     }
 
+    /// $\mathsf{SinsemillaHashToPoint}$ from [§ 5.4.1.9][concretesinsemillahash].
+    ///
+    /// [concretesinsemillahash]: https://zips.z.cash/protocol/nu5.pdf#concretesinsemillahash
+    pub fn hash_to_point(&self, msg: impl Iterator<Item = bool>) -> CtOption<pallas::Point> {
+        self.M.hash_to_point(msg)
+    }
+
+    /// Returns `SinsemillaCommit_r(personalization, msg) = hash_point + [r]R`
+    /// where `SinsemillaHash(personalization, msg) = hash_point`
+    /// and `R` is derived from the `personalization`.
+    #[allow(non_snake_case)]
+    pub fn commit_from_hash_point(
```

Comment:
Why does this need to be exposed? The caller could equivalently clone their `msg` and provide it to both `Self::commit` and `Self::low_level_hash_to_point`. It would also be safer, as it wouldn't be exposing a capability to derive a commitment from the `-r` domain that is inconsistent with the hash point's `-M` domain.

The only reason I can think of to expose this would be performance, and I think any performance concerns here are negligible (or should be backed up with benchmarks showing why the degradation is unacceptable).

Reply:
I agree that the hash_to_point and commit_from_hash_point functions are not essential and potentially unsafe.
So I removed them.